dump/match_infra.py

```diff
@@ -193,6 +199,41 @@ def hget(self, db, key, field):
     def hgetall(self, db, key):
         return self.conn.get_all(db, key)
 
+class RedisPySource(SourceAdapter):
+    """ Concrete Adaptor Class for connecting to APPL_DB using Redis library"""
+
+    def __init__(self, conn_pool, pb_obj):
+        self.conn = None
+        self.pool = conn_pool
+        self.pb_obj = pb_obj
+
+    def connect(self, db, ns):
+        try:
+            self.conn = self.pool.get_dash_conn(ns)
+        except Exception as e:
+            verbose_print("RedisPySource: Connection Failed\n" + str(e))
+            return False
+        return True
+
+    def get_separator(self):
+        return ":"
+
+    def getKeys(self, db, table, key_pattern):
+        bin_keys = self.conn.keys(table + self.get_separator() + key_pattern)
+        return [key1.decode() for key1 in bin_keys]
+
+    def get(self, db, key):
+        key_val  = self.conn.hgetall(key)
+        return get_decoded_value(self.pb_obj, key_val)
```

Comment:
>get_decoded_value [](http://example.com/codeflow?start=15&length=17)

you can `import` immediate before this line, maybe in `__init__` function, and no need to swallow any exception.

Reply:
Done